tests/system/EntityTest.php

```diff
@@ -916,6 +924,28 @@ protected function getSimple()
 		};
 	}
 
+	protected function getSwappedEntity() : Entity
+	{
+		return new class extends Entity
+		{
+			protected $attributes = [
+				'foo' => 'foo',
+				'bar' => 'bar',
+			];
+
+			protected $_original = [
+				'foo' => 'foo',
+				'bar' => 'bar',
+			];
+
+			protected $datamap = [
+				'bar'          => 'foo',
+				'foo'          => 'bar',
+				'original_bar' => 'bar',
```

Comment:
@najdanovicivan 
Each property (array key) in the Entity has a value, so if the column `bar` has two properties, 
we need to discard one value when inserting to database.

The `$datamap` seems to be invalid. Am I something wrong?